pkg/controller/combo_controller.go

```diff
@@ -2,36 +2,191 @@ package controller
 
 import (
 	"context"
+	"fmt"
+	"strings"
+	"time"
 
 	"github.com/go-logr/logr"
+	"github.com/operator-framework/combo/api/v1alpha1"
+	combinationPkg "github.com/operator-framework/combo/pkg/combination"
+	templatePkg "github.com/operator-framework/combo/pkg/template"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/validation"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
-
-	"github.com/operator-framework/combo/api/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/source"
 )
 
 type combinationController struct {
 	client.Client
 	log logr.Logger
 }
 
-func (c *combinationController) manageWith(mgr ctrl.Manager, version int) error {
-	c.log = c.log.V(version)
+// manageWith creates a new instance of this controller
+func (c *combinationController) manageWith(mgr ctrl.Manager, verbosity int) error {
+	c.log = c.log.V(verbosity)
+	templateHandler := handler.EnqueueRequestsFromMapFunc(c.mapTemplateToCombinations)
+
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&v1alpha1.Combination{}).
+		Watches(&source.Kind{Type: &v1alpha1.Template{}}, templateHandler).
 		Complete(c)
 }
 
+// mapTemplateToCombinations is responsible for taking the template object and finding all associated
+// combinations that should be requeued. This should only happen whenever a template is changed in someway.
+func (c *combinationController) mapTemplateToCombinations(template client.Object) []reconcile.Request {
+	if template == nil {
+		return nil
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	requests := []reconcile.Request{}
+
+	// Retrieve and validate the template's name
+	templateName := template.GetName()
+	if isValid := validation.IsValidLabelValue(templateName); isValid != nil {
+		c.log.Info(fmt.Sprintf("template referenced in mapTemplateToCombinations call is not valid: %v", isValid))
+		return requests
+	}
+
+	// Build the label selector for querying
+	templateSelector, err := labels.Parse("combo.ReferencedTemplate=" + templateName)
```

Comment:
(not blocking) Can we (or should we?) make this `combo.ReferencedTemplate` a package constant?

Reply:
Yeah, it makes sense to do that in case we would like to change it in the future. Also makes it clear how to add more labels should that occur moving forward.